packages/medusa/src/loaders/index.ts

```diff
@@ -186,6 +222,8 @@ export default async ({
   track("SEARCH_ENGINE_INDEXING_COMPLETED", { duration: searchAct.duration })
 
   if (featureFlagRouter.isFeatureEnabled(IsolateProductDomainFeatureFlag.key)) {
+    mergeModulesConfig(configModule.modules ?? {}, modulesConfig)
```

Comment:
thought: Would it be reasonable for this to live in `MedusaApp` such that developers using `MedusaApp` directly will have this baked in?

Reply:
On second thought, this merging will not be relevant outside of a Medusa project will it? Because we rely on the `medusa-config.js`

Reply:
yeah it is only relevant when being in a medusa project, outside of it it is not and the user pass the modules or they will be read from somewhere, but medusa app at the moment read from the medusa module config file, which I suppose will evolve in the future.

Reply:
Got it thanks!